clinical/src/test/java/com/hartwig/actin/clinical/curation/CurationDatabaseValidatorTest.java

```diff
@@ -0,0 +1,89 @@
+package com.hartwig.actin.clinical.curation;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import com.hartwig.actin.clinical.curation.config.IntoleranceConfig;
+import com.hartwig.actin.clinical.curation.config.NonOncologicalHistoryConfig;
+import com.hartwig.actin.clinical.curation.config.PrimaryTumorConfig;
+import com.hartwig.actin.clinical.curation.config.SecondPrimaryConfig;
+import com.hartwig.actin.clinical.curation.config.TestCurationConfigFactory;
+import com.hartwig.actin.clinical.datamodel.TestPriorOtherConditionFactory;
+import com.hartwig.actin.clinical.datamodel.TestPriorSecondPrimaryFactory;
+import com.hartwig.actin.doid.DoidModel;
+import com.hartwig.actin.doid.TestDoidModelFactory;
+
+import org.junit.Test;
+
+public class CurationDatabaseValidatorTest {
+
+    @Test
+    public void doesNotCrashOnTestCurationDatabase() {
+        CurationDatabaseValidator validator = TestCurationFactory.createMinimalTestCurationDatabaseValidator();
+
+        validator.validate(TestCurationFactory.createTestCurationDatabase());
+    }
+
+    @Test
+    public void canIdentifyInvalidPrimaryTumorConfigs() {
+        DoidModel doidModel = TestDoidModelFactory.createWithOneParentChild(CurationDatabaseValidator.CANCER_PARENT_DOID, "child");
+
+        PrimaryTumorConfig valid = TestCurationConfigFactory.primaryTumorConfigBuilder().addDoids("child").build();
+        assertTrue(CurationDatabaseValidator.validatePrimaryTumorConfigs(Lists.newArrayList(valid), doidModel));
+
+        PrimaryTumorConfig invalid = TestCurationConfigFactory.primaryTumorConfigBuilder().addDoids("invalid").build();
+        assertFalse(CurationDatabaseValidator.validatePrimaryTumorConfigs(Lists.newArrayList(invalid), doidModel));
+    }
+
+    @Test
+    public void canIdentifyInvalidSecondPrimaryConfigs() {
```

Comment:
Should we add a test case where `priorSecondPrimary.curated()` is null?

Reply:
Agreed, good catch. Missed this scenario (good argument for Optional!)